### DISCUSSION TITLE
With send support, proxy setup examples should be updated

### DISCUSSION THREAD

**Discussion by swstim** [dani-garcia/vaultwarden#1593] [Ideas]
I mentioned this in discussions, but I think it's important enough to post again here:

Now with send support, the [Apache reverse proxy suggested setup](https://github.com/dani-garcia/bitwarden_rs/wiki/Proxy-examples) should be updated.
I have tested and now use the following mod_rewrite config:
```
    RewriteEngine On
    RewriteCond %{HTTP:Upgrade} websocket [NC]
    RewriteCond %{HTTP:Connection} upgrade [NC]
    RewriteRule ^/?(.*) "ws://127.0.0.1:3012/$1" [P,L]
```
This should also be more future proof then specifying a specific path eg `/notifications/hub(.*)`.

_Originally posted by @swstim in https://github.com/dani-garcia/bitwarden_rs/discussions/1230#discussioncomment-532785_

**Comment by pepa65** [comment]
How about the recommended Caddy2 config??

**Comment by jjlin** [comment]
First, this is not a software issue; please follow the instructions given when opening issues. Second, wiki pages can be updated by anyone logged into GitHub, so you can make updates yourself.

That said, Send support doesn't require any changes to the reverse proxy config, so I'm not sure why you seem to think it does. While the existing example in the wiki seems like it should work as is, neither it nor your proposed update are optimal. Since `/notifications/hub` is the only WebSocket path, using `ProxyPassMatch` as shown in https://httpd.apache.org/docs/2.4/mod/mod_proxy_wstunnel.html should be the simplest and most efficient.

**Reply by swstim** [reply]
Apologies, I messed up.

**Reply by jjlin** [reply]
No worries, and sorry if that came across as overly harsh. But feel free to make any improvements to the wiki pages as needed.
